src/platforms/common/blackpill-f4/usbdfu.c

```diff
@@ -22,40 +22,38 @@
 #include <libopencm3/stm32/rcc.h>
 #include <libopencm3/stm32/gpio.h>
 #include <libopencm3/cm3/scb.h>
+#include <libopencm3/usb/dwc/otg_fs.h>
 
 #include "usbdfu.h"
 #include "general.h"
 #include "platform.h"
 
 uintptr_t app_address = 0x08004000U;
-extern uint32_t _ebss; // NOLINT(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp)
+volatile uint32_t magic[2] __attribute__((section(".noinit")));
 
 void dfu_detach(void)
 {
 	scb_reset_system();
 }
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Warray-bounds"
-
 int main(void)
 {
-	volatile uint32_t *magic = (uint32_t *)&_ebss;
 	rcc_periph_clock_enable(RCC_GPIOA);
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Warray-bounds"
-	if (gpio_get(GPIOA, GPIO0) || (magic[0] == BOOTMAGIC0 && magic[1] == BOOTMAGIC1)) {
+	/* Blackpill board has a floating button on PA0. Pull it up and use as active-low. */
+	gpio_mode_setup(GPIOA, GPIO_MODE_INPUT, GPIO_PUPD_PULLUP, GPIO0);
+
+	if (!gpio_get(GPIOA, GPIO0) || (magic[0] == BOOTMAGIC0 && magic[1] == BOOTMAGIC1)) {
 		magic[0] = 0;
 		magic[1] = 0;
-	} else
+	} else {
```

Comment:
I added them because (I touch adjacent code and) they are necessary per #428 and https://www.kernel.org/doc/html/latest/process/coding-style.html#placing-braces-and-spaces , or are you not following this style guide anymore? There's no document on style guide in BMP repo, but `src/target/stm32l4.c:stm32l4_attach()` exhibits both styles.

Reply:
No, and we haven't been for ages. #428 is an old PR (pre 1b² taking the reins on the project) and our advice is consistent with the last 2 years of development.

Files exhibiting both styles is because it's been incrementally applied and if we or someone else has had no reason to touch lines, they haven't yet been updated. Additionally, if you look at the history of that specific file, you'll see we removed those braces ourself when we did the clang-tidy cleanup on that section of that file.

Reply:
We will note that the website incorrectly points to the kernel style, however the needs of this project diverged from the Linux kernel's, so that needs updating. Do note that the website (which the repo does link to) supplements in-repository documentation just like the Linux kernel website does for the kernel.

Reply:
Braces change reverted.
Linux kernel website documentation is generated from in-tree sources, and I couldn't find the BM website sources in this repo. Maybe they're in a neighbouring repo in this GH org?...

Reply:
[yes](https://github.com/blackmagic-debug/black-magic-org), the one clearly labelled as such